javascript/javascript2/week2/script.js

```diff
@@ -0,0 +1,55 @@
+const phones = [
+  { name: "iPhone 14 Pro Max", price: 11550 },
+  { name: "iPhone 14 Pro", price: 10650 },
+  { name: "iPhone 14 Plus", price: 8850 },
+  { name: "iPhone 14", price: 8500 },
+  { name: "Pixel 7 Pro", price: 7600 },
+  { name: "Pixel 7", price: 5200 },
+  { name: "Pixel 6a", price: 4500 },
+  { name: "Galaxy S22 Ultra", price: 7950 },
+  { name: "Galaxy S22", price: 4150 },
+  { name: "Galaxy Z Fold 4", price: 12400 },
+  { name: "Galaxy Z Flip 4 ", price: 8450 },
+  { name: "Oneplus 10 Pro", price: 6150 },
+  { name: "Oneplus 9 Pro", price: 5600 },
+];
+const nameInput = document.querySelector("#searchName");
+const priceInput = document.querySelector("#searchPrice");
+
+const ul = document.querySelector("ul");
+
+nameInput.addEventListener("input", filterdevices);
+renderDevices(phones);
+
+function filterdevices() {
+  const inputText = nameInput.value.toLowerCase();
+
+  const filteredDevices =
+    inputText.trim().length === 0
+      ? phones
+      : phones.filter((device) =>
+          device.name.toLowerCase().includes(inputText)
+        );
+  renderDevices(filteredDevices);
+}
+
+function renderDevices(e) {
+  ul.innerHTML = "";
```

Comment:
good job Malik. 
it's always nice to use described names for variables and parameters . and here when I saw this e at first time I just thought it was an event you want to handle. so I think it's better to name the parameter fx phonesArr or just phones it will make more sense